bundles/org.openhab.binding.automower/src/main/java/org/openhab/binding/automower/internal/things/AutomowerHandler.java

```diff
@@ -243,26 +279,48 @@ public void sendAutomowerCommand(AutomowerCommand command, long commandDurationM
         updateAutomowerState();
     }
 
+    private String restrictedState(RestrictedReason reason) {
+        return "RESTRICTED_" + reason.name();
+    }
+
     private void updateChannelState(Mower mower) {
         if (isValidResult(mower)) {
-            updateState(CHANNEL_MOWER_NAME, new StringType(mower.getAttributes().getSystem().getName()));
-
             updateState(CHANNEL_STATUS_MODE, new StringType(mower.getAttributes().getMower().getMode().name()));
             updateState(CHANNEL_STATUS_ACTIVITY, new StringType(mower.getAttributes().getMower().getActivity().name()));
-            updateState(CHANNEL_STATUS_STATE, new StringType(mower.getAttributes().getMower().getState().name()));
 
-            Instant statusTimestamp = Instant.ofEpochMilli(mower.getAttributes().getMetadata().getStatusTimestamp());
+            if (mower.getAttributes().getMower().getState() != State.RESTRICTED) {
+                updateState(CHANNEL_STATUS_STATE, new StringType(mower.getAttributes().getMower().getState().name()));
+            } else {
+                updateState(CHANNEL_STATUS_STATE,
+                        new StringType(restrictedState(mower.getAttributes().getPlanner().getRestrictedReason())));
+            }
+
             updateState(CHANNEL_STATUS_LAST_UPDATE,
-                    new DateTimeType(ZonedDateTime.ofInstant(statusTimestamp, ZoneId.systemDefault())));
+                    new DateTimeType(toZonedDateTime(mower.getAttributes().getMetadata().getStatusTimestamp())));
             updateState(CHANNEL_STATUS_BATTERY, new QuantityType<Dimensionless>(
                     mower.getAttributes().getBattery().getBatteryPercent(), SmartHomeUnits.PERCENT));
 
             updateState(CHANNEL_STATUS_ERROR_CODE, new DecimalType(mower.getAttributes().getMower().getErrorCode()));
 
-            Instant errorCodeTimestamp = Instant.ofEpochMilli(mower.getAttributes().getMower().getErrorCodeTimestamp());
-            updateState(CHANNEL_STATUS_ERROR_TIMESTAMP,
-                    new DateTimeType(ZonedDateTime.ofInstant(errorCodeTimestamp, ZoneId.systemDefault())));
+            long errorCodeTimestamp = mower.getAttributes().getMower().getErrorCodeTimestamp();
+            if (errorCodeTimestamp == 0L) {
+                updateState(CHANNEL_STATUS_ERROR_TIMESTAMP, UnDefType.NULL);
```

Comment:
Thanks for fixing the problem with the error timestamp :-)

Reply:
No problem :)